process.c

```diff
@@ -5842,25 +5881,25 @@ rb_getpwdirnam_for_login(VALUE login_name)
         }
 
         rb_str_modify_expand(getpwnm_tmp, bufsizenm);
-        bufnm = RSTRING_PTR(getpwnm_tmp);
-        bufsizenm = rb_str_capacity(getpwnm_tmp);
+        args.buf = RSTRING_PTR(getpwnm_tmp);
+        args.bufsize = rb_str_capacity(getpwnm_tmp);
```

Comment:
`long` to `size_t` assignment, as well.

Reply:
Found this affected additional cases, so I added casts for those calls as well.